firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleElement.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.bundle;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public abstract class BundleElement {
```

Comment:
Is this corresponded to `SizedBundleElement` in JS, and the "size" part is not implemented yet. But it seems like `BundlerReader` here directly provides `bytesRead`.

If it is intended as I understand it, I'd prefer to making it an mark interface, and move `fromJson` to `BundleReader`.

Reply:
It is meant to be like the sized element, but I think it is cleaner if it doesn't carry the size information in the Java port. This allows me to do instanceof checks. Instead, the BundleReader has separate state to expose the number of bytes read.